src/objects/utils/autoschema.py

```diff
@@ -1,31 +1,111 @@
 from django.utils.translation import ugettext_lazy as _
 
-from drf_yasg import openapi
-from vng_api_common.inspectors.view import AutoSchema as _AutoSchema
+from drf_spectacular.openapi import AutoSchema as _AutoSchema
+from drf_spectacular.utils import OpenApiParameter
+from vng_api_common.geo import DEFAULT_CRS, HEADER_ACCEPT, HEADER_CONTENT
+
+from objects.api.mixins import GeoMixin
 
 from .serializers import DynamicFieldsMixin
 
 
 class AutoSchema(_AutoSchema):
-    def has_dynamic_fields(self):
-        if self.method != "GET":
-            return False
+    def get_operation_id(self):
+        """
+        Use model name as a base for operation_id
+        """
+        if hasattr(self.view, "queryset"):
+            model_name = self.view.queryset.model._meta.model_name
+            return f"{model_name}_{self.view.action}"
+        return super().get_operation_id()
 
-        serializer = self.get_request_serializer() or self.get_view_serializer()
-        if not serializer:
-            return False
+    def get_override_parameters(self):
+        """ Add request GEO headers"""
+        geo_headers = self.get_geo_headers()
+        content_type_headers = self.get_content_type_headers()
+        field_params = self.get_fields_params()
+        return geo_headers + content_type_headers + field_params
 
-        return isinstance(serializer, DynamicFieldsMixin)
+    def get_geo_headers(self) -> list:
+        if not isinstance(self.view, GeoMixin):
+            return []
+
+        request_headers = []
+        if self.method != "DELETE":
+            request_headers.append(
+                OpenApiParameter(
+                    name=HEADER_ACCEPT,
+                    type=str,
+                    location=OpenApiParameter.HEADER,
+                    required=False,
+                    description=_(
+                        "The desired 'Coordinate Reference System' (CRS) of the response data. "
+                        "According to the GeoJSON spec, WGS84 is the default (EPSG: 4326 "
+                        "is the same as WGS84)."
+                    ),
+                    enum=[DEFAULT_CRS],
+                )
+            )
 
-    def add_manual_parameters(self, parameters):
-        base = super().add_manual_parameters(parameters)
+        if self.method in ("POST", "PUT", "PATCH"):
+            request_headers.append(
+                OpenApiParameter(
+                    name=HEADER_CONTENT,
+                    type=str,
+                    location=OpenApiParameter.HEADER,
+                    required=True,
+                    description=_(
+                        "The 'Coordinate Reference System' (CRS) of the request data. "
+                        "According to the GeoJSON spec, WGS84 is the default (EPSG: 4326 "
+                        "is the same as WGS84)."
+                    ),
+                    enum=[DEFAULT_CRS],
+                ),
+            )
+
+        response_headers = [
```

Comment:
I'm missing the `API-version` header in OAS responses. Is this actually included in API responses?

Reply:
No, since api versioning is turned off now. The old schema had it because it used the same generator as Openzaak, where `API-version` header exists